lib/Doctrine/MongoDB/Aggregation/Builder.php

```diff
@@ -0,0 +1,262 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\MongoDB\Aggregation;
+
+use Doctrine\MongoDB\Collection;
+use Doctrine\MongoDB\Iterator;
+use Doctrine\MongoDB\Query\Expr as QueryExpr;
+use GeoJson\Geometry\Point;
+
+/**
+ * Fluent interface for building aggregation pipelines.
+ *
+ * @author alcaeus <alcaeus@alcaeus.org>
+ */
+class Builder
+{
+    /**
+     * The Collection instance.
+     *
+     * @var Collection
+     */
+    protected $collection;
+
+    /**
+     * @var Stage[]
+     */
+    protected $stages;
+
+    /**
+     * Create a new query builder.
+     *
+     * @param Collection $collection
+     */
+    public function __construct(Collection $collection)
+    {
+        $this->collection = $collection;
+    }
+
+    /**
+     * @return Expr
+     */
+    public function expr()
+    {
+        return new Expr();
+    }
+
+    /**
+     * Executes the aggregation pipeline
+     *
+     * @param array $options
+     * @return Iterator
+     */
+    public function execute($options = array())
+    {
+        return $this->collection->aggregate($this->getPipeline(), $options);
+    }
+
+    /**
+     * Outputs documents in order of nearest to farthest from a specified point. You can only use this as the first stage of a pipeline.
+     *
+     * A GeoJSON point may be provided as the first and only argument for
+     * 2dsphere queries. This single parameter may be a GeoJSON point object or
+     * an array corresponding to the point's JSON representation. If GeoJSON is
+     * used, the "spherical" option will default to true.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/geoNear/
+     *
+     * @param float|array|Point $x
+     * @param float $y
+     * @return Stage\GeoNear
+     */
+    public function geoNear($x, $y = null)
+    {
+        return $this->addStage(new Stage\GeoNear($this, $x, $y));
+    }
+
+    /**
+     * Returns a certain stage from the pipeline
+     *
+     * @param integer $index
+     * @return Stage
+     */
+    public function getStage($index)
+    {
+        if (!isset($this->stages[$index])) {
+            throw new \InvalidArgumentException("Could not find stage with index {$index}.");
+        }
+
+        return $this->stages[$index];
+    }
+
+    /**
+     * Returns the assembled aggregation pipeline
+     *
+     * @return array
+     */
+    public function getPipeline()
+    {
+        $pipeline = array();
+
+        foreach ($this->stages as $stage) {
+            $pipeline[] = $stage->getExpression();
+        }
+
+        return $pipeline;
+    }
+
+    /**
+     * Groups documents by some specified expression and outputs to the next stage a document for each distinct grouping.
```

Comment:
Similar comment as with `geoNear()`. Even if one sentence, we can wrap this.

I'll refain from leaving redundant comments on the other docblocks below.